### PR TITLE
[#64] [#66] [Android] [UI] [Integrate] As a user, I can see form answers

### LINKED ISSUE #64: [Android] [UI] As a user, I can see form answers
## Why
A textfield question is a question that allow users to answer with multiple short answers. There can be several textfields in a single textfield question. It is commonly used to ask for user's information.

### Example questions
- Don't miss out on our Exclusive Promotions!
    - Textfield 1: Name
    - Textfield 2: Email
    - Textfield 3: Mobile no.
- Your contact information
    - Textfield 1: Name
    - Textfield 2: Room no.
    - Textfield 3: Email address
    - Textfield 4: Phone

## Acceptance Criteria
For textfield questions,
- Display an inline textfield
    - Display a placeholder in the textfield
    - When a textfield is not empty, increase the opacity
- Display the keyboard when a textfield is selected
    - Tap anywhere outside of the textfield to release the keyboard
    - The keyboard must not cover the selected textfield

## Design

[Textfield Question](https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge-Mobile-App?node-id=2632%3A519)

### LINKED ISSUE #66: [Android] [Integrate] As a user, I can see form answers
## Why

A textfield question is a question that allow users to answer with multiple short answers. There can be several textfields in a single textfield question. It is commonly used to ask for user's information.

### Example questions
Don't miss out on our Exclusive Promotions!
- Textfield 1: Name
- Textfield 2: Email
- Textfield 3: Mobile no.

Your contact information
- Textfield 1: Name
- Textfield 2: Room no.
- Textfield 3: Email address
- Textfield 4: Phone

## Acceptance Criteria
When a Question model has `display_type` as `textfield`,
- Display a textfield question
- Use the answers to show as textfields
    - Apply the answer's text as its textfield placeholder

## Design

[Textfield Question](https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge-Mobile-App?node-id=2632%3A519)

### PR DESCRIPTION
**Pull request by suho**
- Resolves #64
- Resolves #66

## What happened 👀

Added form answers for the survey questions.

## Insight 📝

`N/A`

## Proof Of Work 📹

Screenshot 2022-12-15 at 11 10 22

### CODE REVIEW

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/FormAnswer.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        modifier = modifier
+    ) {
+        placeholders.forEachIndexed { index, placeholder ->
+            var value by remember { mutableStateOf("") }
+            val focusManager = LocalFocusManager.current
+            TextField(
+                value = value,
+                onValueChange = {
+                    value = it
+                    onTextChange(Pair(index, it))
```
Comment (hoangnguyen92dn) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/FormAnswer.kt:38]: JIDYDK, `Pair(index, it)` ~ `index to it`

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/surveyquestions/SurveyQuestionsScreen.kt [hunk 2]
```diff
@@ -276,6 +276,11 @@ private fun Answer(type: QuestionDisplayType, onAnswersChange: (List<AnswerInput
             placeholder = type.placeholder,
             modifier = Modifier.height(170.dp)
         )
+        is QuestionDisplayType.Textfield -> FormAnswer(
+            placeholders = type.placeholders,
+            onTextChange = {
+                onAnswersChange(listOf(AnswerInput.Content(index = it.first, content = it.second)))
+            })
```
Comment (hoangnguyen92dn) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/surveyquestions/SurveyQuestionsScreen.kt:283]: Could you please try this? 😄 It's better for readability IMO

```suggestion
            onTextChange = { (index, content) ->
                onAnswersChange(listOf(AnswerInput.Content(index = index, content = content)))
            })
```

**Review by hoangnguyen92dn — COMMENTED** [review]
The rest LGTM 🚀

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/surveyquestions/SurveyQuestionsScreen.kt (hunk 2 above)
Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/surveyquestions/SurveyQuestionsScreen.kt]: Good to know this! Fixed in https://github.com/suho/kmm-ic/pull/120/commits/3ac803117d682a8efb02d784f7c71974f3358abd

**Review by hoangnguyen92dn — APPROVED** [review]
LGTM 🚀

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/FormAnswer.kt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import androidx.compose.ui.unit.dp
+import co.nimblehq.ic.kmm.suv.android.ui.theme.AppTheme
+import co.nimblehq.ic.kmm.suv.android.ui.theme.Typography
+
+typealias Placeholder = String
+
+@Composable
+fun FormAnswer(
+    placeholders: List<Placeholder>,
+    onTextChange: (Pair<Int, String>) -> Unit,
```
Comment (luongvo) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/FormAnswer.kt:24]: `Pair<Int, String>` is too value, we need a DTO

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/FormAnswer.kt [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                    Text(
+                        text = placeholder,
+                        style = Typography.subtitle1
+                    )
+                },
+                colors = formTextFieldColors(value.isEmpty()),
+                keyboardActions = KeyboardActions(onDone = { focusManager.clearFocus() }),
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .size(56.dp)
```
Comment (luongvo) [inline android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/FormAnswer.kt:51]: it seems we have a dimension for this value: defaultComponentHeight

**Review by luongvo — COMMENTED** [review]
rest lgtm

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/FormAnswer.kt (hunk 3 above)
Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/FormAnswer.kt:24]: Please ignore this, I've changed on all answers for consistency: https://github.com/suho/kmm-ic/pull/123/files#diff-0a5f71ca5c346d1a7908daa87906e87bd8d038716044a0a7b5526578016255e1R34

File: android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/FormAnswer.kt (hunk 4 above)
Comment (suho) [reply android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/components/FormAnswer.kt]: Fixed in https://github.com/suho/kmm-ic/pull/120/commits/2ce21e00f1e35f205215122019f7b0a0e65747f7

**Review by luongvo — APPROVED** [review]

**Pull request merged**